lib/tasks/publishing_api.rake

```diff
@@ -1,29 +1,38 @@
 require "gds_api/publishing_api/special_route_publisher"
 
 namespace :publishing_api do
-  namespace :draft do
-    namespace :populate do
-      desc "export all whitehall content to draft environment of publishing api"
-      task :all => :environment do
-        Whitehall::PublishingApi::DraftEnvironmentPopulator.new(logger: Logger.new(STDOUT)).call
+  namespace :republish do
+    # republish is the correct update_type to use when migrating content
+    # that has been published in Whitehall to the publishing api
+    desc "republish editions to the Publishing API (model_class_name example: CaseStudy)"
+    task :editions, [:model_class_name] => :environment do |t, args|
+      model = args[:model_class_name].constantize
+      model.latest_published_edition.find_each do |edition|
+        Whitehall::PublishingApi.republish_async(edition)
+        print "."
       end
-
-      desc "export Case studies to draft environment of publishing api"
-      task :case_studies => :environment do
-        Whitehall::PublishingApi::DraftEnvironmentPopulator.new(items: CaseStudy.latest_edition.find_each, logger: Logger.new(STDOUT)).call
+      # for any unpublished drafts
+      model.latest_edition.find_each do |edition|
+        Whitehall::PublishingApi.save_draft_async(edition)
+        print "."
       end
     end
-  end
 
-  namespace :live do
-    namespace :populate do
-      desc "export all published whitehall content to live environment of publishing api"
-      task :all => :environment do
-        Whitehall::PublishingApi::LiveEnvironmentPopulator.new(logger: Logger.new(STDOUT)).call
+    desc "republish non-edition content to the Publishing API (model_class_name example: TakePartPage)"
+    task :non_editions, [:model_class_name] => :environment do |t, args|
+      model = args[:model_class_name].constantize
+      model.all.find_each do |instance|
+        Whitehall::PublishingApi.republish_async(instance)
+        print "."
       end
+    end
 
-      task :case_studies => :environment do
-        Whitehall::PublishingApi::LiveEnvironmentPopulator.new(items: CaseStudy.latest_published_edition.find_each, logger: Logger.new(STDOUT)).call
+    desc "generate content ids for non-edition content prior to publishing"
+    task :add_content_ids, [:model_class_name] => :environment do |t, args|
+      model = args[:model_class_name].constantize
+      model.all.find_each do |instance|
+        instance.update_column(:content_id, SecureRandom.uuid)
```

Comment:
This should include a guard for rows that already have content ids, so we don't accidentally overwrite them.